lib/router/utils.js

```diff
@@ -25,19 +25,9 @@ export function merge(hash, other) {
   Extracts query params from the end of an array
 **/
 export function extractQueryParams(array) {
-  let len = array && array.length,
-    head,
-    queryParams;
-
-  if (
-    len &&
-    len > 0 &&
-    array[len - 1] &&
-    array[len - 1].hasOwnProperty('queryParams')
-  ) {
-    queryParams = array[len - 1].queryParams;
-    head = slice.call(array, 0, len - 1);
-    return [head, queryParams];
+  let possibleQuery = array[array.length - 1];
```

Comment:
The prior version supported `array` being `undefined`, and this one doesn't.  Are we 100% certain that all callsites pass a valid array?

Reply:
Also, isn't this actually `possibleOptions` (since the "query params" object is nested inside of it)?